### PR TITLE
Allow Aggregations in Case Expressions

### PR DESCRIPTION
**Pull request by ege-st**
(Bugfix)

In Calcite, this removes the check that prevents aggregation functions from being used within a Case/When expression in the V1 engine.

Testing:
1. Added a unit test for V2 engine
2. Validated locally with Pinot running the query: `select AirlineID, (CASE WHEN sum(ActualElapsedTime) > 0 THEN sum(ActualElapsedTime) ELSE 0 END) AS cpm from airlineStats group by AirlineID` and comparing the results between V1 and V2 engine.

### CODE REVIEW

**Comment by ege-st** [comment]
Could not find the right place to put a V1 engine unit test for this change, so please let me know where it should go.

File: pinot-query-runtime/src/test/java/org/apache/pinot/query/QueryTestSet.java [hunk 1]
```diff
@@ -106,6 +106,9 @@ public Object[][] provideTestSql() {
             + " SUM(CASE WHEN a.col2 <> 'foo' AND a.col2 <> 'alice' THEN 1 ELSE 0 END) as unmatch_sum "
             + " FROM a WHERE a.ts >= 1600000000 GROUP BY a.col1"},
 
+        new Object[]{"SELECT a.col1, CASE WHEN sum(a.col3) = 0 THEN 0 ELSE SUM(a.col3) END AS match_sum "
+            + " FROM a WHERE a.ts >= 1600000000 GROUP BY a.col1"},
```
Comment (ege-st) [inline pinot-query-runtime/src/test/java/org/apache/pinot/query/QueryTestSet.java:110]: Would like to add additional unit tests for V2 as I think this test just verifies Semantics and Syntax, would like to test execution results.

Comment (Jackie-Jiang) [reply pinot-query-runtime/src/test/java/org/apache/pinot/query/QueryTestSet.java:110]: Actually we want to add tests for V1. It is already working in V2

**Comment by ege-st** [comment]
Correct. I added a test for v2 because I didn't see one and just wanted to
be thorough.

Hiwever I had a comment that I couldn't find the right place to add the
test for V1. So just need to find out where to write that test and I'll add
it.

**Comment by Jackie-Jiang** [comment]
You may add some queries in `BaseClusterIntegrationTestSet.testHardcodedQueries()` which can apply to both V1 and V2

File: pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java [hunk 2]
```diff
@@ -141,27 +141,6 @@ public void testCaseWhenStatements() {
     Assert.assertEquals(caseFunc.getOperands().get(6).getLiteral().getFieldValue(), 0L);
   }
 
-  @Test(expectedExceptions = SqlCompilationException.class)
-  public void testInvalidCaseWhenStatements() {
```
Comment (gortiz) [inline pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java:145]: Couldn't we reuse the text to verify this works now?

Even something as simple as compiling the same SQL query and assert no failure is found. May not be super useful, but given it is already written, we could use it

**Comment by ege-st** [comment]
That's a very good suggestion, I'll do that.

File: pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java (hunk 2 above)
Comment (ege-st) [reply pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java:145]: Done.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**